txnkv/txnlock/lock_resolver.go

```diff
@@ -448,7 +448,11 @@ func (lr *LockResolver) resolveLocks(bo *retry.Backoffer, opts ResolveLocksOptio
 	var resolve func(*Lock, bool) (TxnStatus, error)
 	resolve = func(l *Lock, forceSyncCommit bool) (TxnStatus, error) {
 		status, err := lr.getTxnStatusFromLock(bo, l, callerStartTS, forceSyncCommit, detail)
-		if err != nil {
+
+		if _, ok := errors.Cause(err).(primaryMismatch); ok && l.LockType == kvrpcpb.Op_PessimisticLock {
```

Comment:
Maybe we could print an `ERROR` or `FATAL` log for the `l.LockType != kvrpcpb.Op_PessimisticLock` path since it's unexpected?

Reply:
I'll add an error here. Actually the error will be returned to TiDB, and TiDB will print it as an unknown error anyway.

Reply:
I noticed that there might be other problem about the primary when fair-locking is used...

Reply:
@MyonKeminta 
The check issue for fair lock is recorded in https://github.com/pingcap/tidb/issues/43540, could we merge this first?

Reply:
> @MyonKeminta The check issue for fair lock is recorded in [pingcap/tidb#43540](https://github.com/pingcap/tidb/issues/43540), could we merge this first?

yes. lets merge it.